### ISSUE TITLE
Collected issues

### ISSUE THREAD

**Issue by Darkwolf1515** [anansi-project/rfcs#21]
How will collected issues be handled? Comic vine handles them pretty awfully, making them their own series rather than apart of the series of issues they're collecting (example, https://comicvine.gamespot.com/transformers/4050-29903/, each and every one of the 7 collected issues have their own series page, and are issue one of their respective series, had to manually edit each one in comictagger) Most add tpb to the end despite it being available digitally, which is also annoying.

I think a solution would be to have a specific tag or option for collected editions, a simple true/false toggle to say if its a collected issue in a series would work well, it would be apart of the same series as its non collected counterparts, but apps could separate them into their own categories, so you could use the same issue # and title fields, its just that one tag would change what its categorized as. (would look like this) 
![image](https://user-images.githubusercontent.com/65789901/92161276-eadd7400-ee1f-11ea-9c4e-6e1bc2be5559.png)

**Comment by gotson** [comment]
I don't quite get your link, it shows 31 issues in a volume, nothing collected about it.

**Comment by Darkwolf1515** [comment]
That’s my point, the collected issues are shoved in the links above, as their own series, rather than as apart of the series they’re collecting.

**Comment by gotson** [comment]
I don't understand what you are trying to say. Can you be more specific? I have no knowledge on those comics, so I may be missing the point.

Can you also describe what you mean exactly by "collected"? Is this something like how trade paper backs contain multiple single issues?

**Comment by Darkwolf1515** [comment]
Yeah that’s it, tbps often become their own series for whatever reason, it’s annoying to deal with when you want them to be organized with non tbps in the same series.

**Comment by gotson** [comment]
It's quoted often that TBPs collect issues from multiple series of the same universe. In that case it makes sense to be their own series.

It's more a source of truth organisation question, which will not impact the metadata format to be honest.

**Comment by Darkwolf1515** [comment]
My issue was when say, I had 7 tpbs from the same single comic series, and each one became their own separate series, rather than grouping together, in the case of mixed series I get it, but otherwise it’s very annoying.

**Comment by theotocopulitos** [comment]
This is terrible in ComicVine, for instance for Marve Epic Collection:

Marvel publishes several "Epic Collection" series... I have seen that in Comicvine, some of these have been implemented as Series (for example, those related to Star Wars, like this one Star Wars Legends Epic Collection: The Empire).

On the other hand, other "Epic Collection" series have been included in the wiki as individual one-shots, when they actually comprise a single volume with several numbered issues. For example, Silver Surfer Epic Collection should be a single series that include the following books as issues, not as they appear now in comicvine, each issue as a separate series:

Issue 1: Silver Surfer Epic Collection: When Calls Galactus #1 - Volume 1 (now belonging to Volume Silver Surfer Epic Collection: When Calls Galactus )

Issue 2: not yet published

Issue 3: Silver Surfer Epic Collection: Freedom #1 - TPB (now in Volume Silver Surfer Epic Collection: Freedom )

Issue 4: not yet published

Issue 5: not yet published

Issue 6: Silver Surfer Epic Collection: Thanos Quest #1 - TPB (now in volume Silver Surfer Epic Collection: Thanos Quest)

Issue 7: not yet published

Issue 8: Silver Surfer Epic Collection: The Infinity Gauntlet #1 - TPB (now in volume Silver Surfer Epic Collection: The Infinity Gauntlet

Issues 9-12: not yet published

Issue 13: Silver Surfer Epic Collection: Inner Demons #1 - Volume 13, (now in Silver Surfer Epic Collection: Inner Demons)

**Comment by gotson** [comment]
At the moment the same story in a TPB and in a single issue will not be linked together, that's because the TPB is one _Book_, and the single issue is another _Book_.

[BookBrainz](https://bookbrainz.org/help) addresses this by abstracting the story as a _Work_, which can be contained in multiple _Editions_.

In that model, there is a single Story, contained in a single issue, and a TPB.

If we had this model, we could link a TPB and single issues.

Is that something we would be interested in?

**Comment by Darkwolf1515** [comment]
My current issue is that, sometimes TPB's will be their own series, rather than apart of the series of books its collecting (excluding when a TBP has multiple series. 

I was hoping that, if an official db was made around this new metadata type, TBP's would be added to existing series, rather than made their own.

**Comment by Darkwolf1515** [comment]
This isnt so much an issue with the format of this new metadata, but one of if you create a db that users can download comic info from. 

Ill try to articulate the issue as best I can.

Say for instance, we have a comic series, we'll call it batdog, batdog has 30 issues total, all of those 30 issues, are apart of the same series, "batdog" this is good, as it groups them all together. Now lets say batdog has some TBP's, the first one, contains issue 1-15, and is called, "Batdog: origins". the second one is called "batdog, the second' With the current way comicvine does things, both of those would be made into their own separate series, titled "batdog origins" and "batdog, the second". This is bad, as now, those tbp's not only wont integrate with eachother, but they wont integrate with the standard comics either.

The solution, of course, is to have the series be "batdog" for all the comics + tbp's with the tbp's titles as well, the tiles of the book. 

Changing the data from 

Series: Batdog: origins
Title: Batdog origins

Series: Batdog, the second
Title: Batdog, the second

To the proper way

Series: Batdog
Title: origins

Series: Batdog
Title: The second

Its long winded, but I hope that describes my issue.

**Comment by gotson** [comment]
> My current issue is that, sometimes TPB's will be their own series, rather than apart of the series of books its collecting (excluding when a TBP has multiple series.
> 
> I was hoping that, if an official db was made around this new metadata type, TBP's would be added to existing series, rather than made their own.

That's 2 different things:
- the people adding data in ComicVine in a not-so-efficient way (1 TPB = 1 serie)
- the data model to represent things properly

Here i cannot fix point 1, but we can make sure we properly model data to cater for various use cases.

---

As for your example, there's 2 way of seeing it (i don't count the shitty CV way to have a TPB = a serie):
- the way i highlighted above, to split the _work_ from the _publication_: that way you can know that a particular work has been published in a series of single issues, and into a TPB
- consider the single issue series and the TPB series (with 2 issues) to be different series: Batdog, and Batdog (TPB)

Point 2 is easy, doesn't change the model, and is in line with the publishing world.

Point 1 is more exhaustive, and you can know which TPB collects which single issues. You can also handle reissues, etc.

**Comment by Darkwolf1515** [comment]
So the options are, make ALL the tbp's their own series, meaning they will group fine, or what I suggested, that's what you said right?

**Comment by gotson** [comment]
I think you misunderstand what a data model can provide, and tour particular bad experience with comicvine.

A data model is a way to model data, if used badly (like CV) you get bad results. It's not the model"s fault. 

Make TPBs as a TPB series is how many tools already do, or even publishers will advertise the tpb as a different series altogether.

If you want to be able to know which tpb collects which single issue, we need a different data model.

**Comment by Darkwolf1515** [comment]
> 
> 
> I think you misunderstand what a data model can provide, and tour particular bad experience with comicvine.
> 
> A data model is a way to model data, if used badly (like CV) you get bad results. It's not the model"s fault.
> 
> Make TPBs as a TPB series is how many tools already do, or even publishers will advertise the tpb as a different series altogether.
> 
> If you want to be able to know which tpb collects which single issue, we need a different data model.

No, my issue will come into play if a db is ever created, I guess I shouldn't have made my post just yet, its not an issue with the data format, just a future db.

**Comment by gotson** [comment]
Then make sure the data model can fit your requirements, the future db will use it.

**Comment by Darkwolf1515** [comment]
> 
> 
> Then make sure the data model can fit your requirements, the future db will use it.

At worst, I just have to manually edit the series like I do with comicvine, its not a huge deal, all I want is for the db to have tbps series, match the series name they're collecting, thats all.

**Comment by timgilbert** [comment]
I do think it would be a good idea to have something like a "collects" attribute on Book which would refer to one or more other Books. So you might have _Batdog Origins TPB_, which could still be its own series if that's how the various sources of truth have set it up, but then it would have a "collects" attribute pointing to the (separate, unique) Books _Batdog Origins 01_, _Batdog Origins 02_, and _Batdog Origins 03_, where each of those Books belongs to the distinct _Batdog Origins_ series, not the _Batdog Origins TPB_ series that the collection itself is in.

This model could handle the more pathological cases, like when a publisher creates an _Infinite Batdogs_ crossover collection which collects issues from several different series (_Batdog_, _Catbird_, _Infinite Batdog Origins_)  into a single TPB. And in theory, this small part of the metadata could reside in a separate, open-source repository which is complementary to ComicVine, so it wouldn't necessarily require either getting sources of truth to change their models, nor reimplementing ComicVine from the ground up.

**Comment by nevven** [comment]
> 
> 
> At the moment the same story in a TPB and in a single issue will not be linked together, that's because the TPB is one _Book_, and the single issue is another _Book_.
> 
> [BookBrainz](https://bookbrainz.org/help) addresses this by abstracting the story as a _Work_, which can be contained in multiple _Editions_.
> 
> In that model, there is a single Story, contained in a single issue, and a TPB.
> 
> If we had this model, we could link a TPB and single issues.
> 
> Is that something we would be interested in?

BookBrainz approach is not bad i think, but do we think of a series as a story? Because if we do than the Single Issues and TPB is the same `series`.

Batdog with 30 issues total
`series`: batdog

Batdog TPB collecting issues 1-15
`title`: batdog origins
`series`: batdog

Because it is, in fact the same thing, just different format

**Comment by Darkwolf1515** [comment]
> > At the moment the same story in a TPB and in a single issue will not be linked together, that's because the TPB is one _Book_, and the single issue is another _Book_.
> > [BookBrainz](https://bookbrainz.org/help) addresses this by abstracting the story as a _Work_, which can be contained in multiple _Editions_.
> > In that model, there is a single Story, contained in a single issue, and a TPB.
> > If we had this model, we could link a TPB and single issues.
> > Is that something we would be interested in?
> 
> BookBrainz approach is not bad i think, but do we think of a series as a story? Because if we do than the Single Issues and TPB is the same `series`.
> 
> Batdog with 30 issues total
> `series`: batdog
> 
> Batdog TPB collecting issues 1-15
> `title`: batdog origins
> `series`: batdog
> 
> Because it is, in fact the same thing, just different format

Exactly what I was asking for, thank you!

**Comment by gotson** [comment]
That's doesn't always work. TPB can collect issues from other series too. For example the Fables TBP collect some issues from Jack of Fables.

**Comment by nevven** [comment]
It doesn't always work that's true, comics have a large amount of fringe cases unfortunately 
Btw i'm very happy that this project started @gotson
In my attempts to catalog my comic collection, i almost always had to manually correct Comicvine, it can be frustrating

**Comment by dankles** [comment]
I think a simple approach would be to allow a book to be part of multiple series. I don't really understand UML, but I think it reads that, a `Book` can belong to `0..1` `Series`. If we changed that to `0..*`, it would basically make a Series a type of "Tag". That allows a Collected Edition/TPB to be "Tagged" as part of multiple series. I like the flexibility in that approach, given so many fringe cases.

**Comment by gotson** [comment]
I think we need to go back to what is a Series. When saying Series, people will usually understand:
- all books part of the series make a whole set. So only single issues, or only the TPBs, but not a mix of both.
- a Series can be complete if all books are published and collected if you have all of them.

We also have bad examples of how series are handled in CV, where for instance each TPB is a series of its own, instead of all the TPBs forming their own Series.

I do agree though that it would be interesting to have the link between different editions of the same series, ie Single Issues vs TPBs.

We could introduce the concept of Edition, and have an Edition being the link between a Series and its Books.

You would have for the Batdog example above
- a single Batdog Series
- 2 editions linked to that Series:
  - Single Issues: with 30 books
  - TPBs: with 1 book

That doesn't solve the problem of TPBs that collect issues from various Series, but i think in that case there is always a _dominant_ Series, in the example i gave, it's Fables. Or we could also have a `0..n` link between Series and Edition, so an Edition could be attached to multiple Series, in that case Fables and Jack Of Fables.

Back to the BookBrainz work thing, i think it could be very well adapted to handle covers, which were mentioned in #18. By having each cover being a separate work, we can have the original date of the Work itself, and the published date of the Book.

---

Something i wanted to do soon was to "test" the model on various use cases, ie having real world examples on how the data would be like. I think it would be good to test the model on complex cases.

**Comment by jakem742** [comment]
As you mentioned, it really comes down to how series get handled. I personally like the idea of Editions of a particular series, where the series describes the set of story events and the editions collects a particular mediaType or set of published books.

**Series Only Model** (Series object contains all media types)
If a TPB has issues from several different series, these could all be represented with a seriesIssue entry for each item in the TPB. You could then have multiple files each referring to a specific seriesIssue (eg you might have a book which is the individual issue, a book which is the TPB, and a book which is an even bigger omnibus, all of which contain the same seriesIssue). Telkine has already mentioned this [here](https://github.com/anansi-project/rfcs/issues/21#issuecomment-848037366)

My proposal would be to have a mediaType associated with each book, and then a book is associated with 1 or more series. The user's software could then filter books in a series by mediaFormat to get only the books of a certain type. 

I have mentioned this concept in (#40).

____

**Edition > Series model** (Separate editions for each media type / publication run)
If each series is an edition with each book being uniquely tied to a single edition, the seriesIssue field only really describes that books place in the edition. It doesn't allow for  references to which story events are contained. eg. You might have a TPB which is Vol. 2 (so seriesIssue number = 2 for that book). 

It would be extremely helpful in these circumstances to also tie that TPB to specific story events. I imagine issues would be the typical way to refer to these events, as they are the base unit for story events. So each TPB book could also have references to Series "issues" either as a direct relationship with the issue's "Book" object, or through an independent list of objects associated with the Series object. Having a set of events associated with the series objects makes the most sense to me, as this allows all Book objects to be related to 1 or more events, regardless of whether they're an issue, TPB, Omni, etc. 

So basically Book --> Edition (published set) --> Series (story). The Series object contains a list of numbered events (correlating to issues, I imagine). Each book contains a relationship with an Edition, and a numbered list of the events it contains (either through relationship with the series, or simply as a set of numbers).

**Comment by baodrate** [comment]
> That doesn't solve the problem of TPBs that collect issues from various Series, but i think in that case there is always a _dominant_ Series

I don't think that's a safe assumption. There are many cases (crossover events/special event books e.g. holidays/etc.) where there is no "dominant" series or it is not one of the series whose issues is contained within.
